### Importing Required Libraries

In [ ]:
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Embedding, Conv1D, GlobalMaxPooling1D, Dense, Dropout

### CNN Model

In [24]:
model = Sequential([
    Embedding(input_dim=10000, output_dim=64),
    Conv1D(filters=128, kernel_size=5, activation='relu'),
    GlobalMaxPooling1D(),
    Dropout(0.5),
    Dense(10, activation='relu'),
    Dense(1, activation='sigmoid') 
])

model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])
model.summary()

Model: "sequential"

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
┃ Layer (type)                         ┃ Output Shape                ┃         Param # ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
│ embedding (Embedding)                │ ?                           │     0 (unbuilt) │
├──────────────────────────────────────┼─────────────────────────────┼─────────────────┤
│ conv1d (Conv1D)                      │ ?                           │     0 (unbuilt) │
├──────────────────────────────────────┼─────────────────────────────┼─────────────────┤
│ global_max_pooling1d                 │ ?                           │               0 │
│ (GlobalMaxPooling1D)                 │                             │                 │
├──────────────────────────────────────┼─────────────────────────────┼─────────────────┤
│ dropout (Dropout)                    │ ?                           │               0 │
├──────────────────────────────────────┼─────────────────────────────┼─────────────────┤
│ dense (Dense)                        │ ?                           │     0 (unbuilt) │
├──────────────────────────────────────┼─────────────────────────────┼─────────────────┤
│ dense_1 (Dense)                      │ ?                           │     0 (unbuilt) │
└──────────────────────────────────────┴─────────────────────────────┴─────────────────┘

 Total params: 0 (0.00 B)

 Trainable params: 0 (0.00 B)

 Non-trainable params: 0 (0.00 B)

### Train the Model

In [ ]:
history = model.fit(
    X_train, y_train,
    epochs=15,             
    batch_size=128,
    validation_split=0.3,
    verbose=1)

Epoch 1/15
192/192 ━━━━━━━━━━━━━━━━━━━━ 20s 81ms/step - accuracy: 0.6943 - loss: 0.5613 - val_accuracy: 0.8416 - val_loss: 0.3660
Epoch 2/15
192/192 ━━━━━━━━━━━━━━━━━━━━ 17s 89ms/step - accuracy: 0.8562 - loss: 0.3370 - val_accuracy: 0.8612 - val_loss: 0.3245
Epoch 3/15
192/192 ━━━━━━━━━━━━━━━━━━━━ 16s 84ms/step - accuracy: 0.9095 - loss: 0.2315 - val_accuracy: 0.8670 - val_loss: 0.3232
Epoch 4/15
192/192 ━━━━━━━━━━━━━━━━━━━━ 17s 90ms/step - accuracy: 0.9452 - loss: 0.1511 - val_accuracy: 0.8604 - val_loss: 0.3611
Epoch 5/15
192/192 ━━━━━━━━━━━━━━━━━━━━ 20s 85ms/step - accuracy: 0.9694 - loss: 0.0925 - val_accuracy: 0.8549 - val_loss: 0.4290
Epoch 6/15
192/192 ━━━━━━━━━━━━━━━━━━━━ 17s 90ms/step - accuracy: 0.9823 - loss: 0.0582 - val_accuracy: 0.8547 - val_loss: 0.4987
Epoch 7/15
192/192 ━━━━━━━━━━━━━━━━━━━━ 15s 80ms/step - accuracy: 0.9900 - loss: 0.0351 - val_accuracy: 0.8517 - val_loss: 0.5749
Epoch 8/15
192/192 ━━━━━━━━━━━━━━━━━━━━ 17s 86ms/step - accuracy: 0.9928 - loss: 0.0261 - 

### Evaluate the Model

In [ ]:
loss, accuracy = model.evaluate(X_test, y_test)
print(f"Test Accuracy: {accuracy:.2f}")

## Summary of Results:
#### - Model Architecture: Embedding → CNN → Dropout → Dense layers
#### - Dataset Size: ~50,000 samples
#### - Data Split: 70% training, 30% testing
#### - Training Setup: 15 epochs, batch size = 128, validation_split = 0.3
#### - Accuracy: **85.5%**